### PR TITLE
[4.0] Move language strings for dashboard titles to respective component.

### PR DESCRIPTION
**Pull request by Bakual**
Inspired by discussion in https://github.com/joomla/joomla-cms/pull/31382

Currently, the component specific dashboards of com_menus, com_content, com_users and com_privacy take their dashboard title from a string in com_cpanel.
This makes no sense and is a special threatment of core components which shouldn't happen.

Also current code requires that 3rd party dashboard are named as "com_foo" unlike core dashboards which are just named "foo". This PR will work with both variants for 3rd party.

### Summary of Changes
Moving the strings to the correct component language file
Adjusting the title lookup code so it first tries to load the string from a component language file. It prepends the dashboard name with "com_" (eg "content" becomes "com_content) if not already present to guess the related component.
If no string is found it will try with a com_cpanel string.

I removed the string COM_CPANEL_DASHBOARD_CONTENT_WORKFLOW_TITLE="Content Workflow Dashboard" as I couldn't find a workflow related dashboard.

### Testing Instructions
Test the various dashboards if the title is still the same.

### Actual result BEFORE applying this Pull Request
Titles are shown, but taken from com_cpanel

### Expected result AFTER applying this Pull Request
In core all looks fine, titles come from the component language files

### Documentation Changes Required
Maybe if dashboard documentation exists somewhere.

### CODE REVIEW

**Comment by Bakual** [comment]
On a sidenote, the icon code is very strange as well, using a language string to define the icon used. However language strings are for translating purposes, not for design.
For core this was already changed to a hardcoded list. Which is better but still not good and not useable for 3rd parties.

If someone has a good idea how to solve this, I'm all ears.

File: administrator/language/en-GB/com_cpanel.ini [hunk 1]
```diff
@@ -8,13 +8,8 @@ COM_CPANEL_ADD_DASHBOARD_MODULE="Add module to the dashboard"
 COM_CPANEL_ADD_MODULE_MODAL_TITLE="Add Module"
 COM_CPANEL_DASHBOARD_BASE_TITLE="Home Dashboard"
 COM_CPANEL_DASHBOARD_COMPONENTS_TITLE="Components Dashboard"
-COM_CPANEL_DASHBOARD_CONTENT_TITLE="Content Dashboard"
-COM_CPANEL_DASHBOARD_CONTENT_WORKFLOW_TITLE="Content Workflow Dashboard"
```
Comment (infograf768) [inline administrator/language/en-GB/com_cpanel.ini:12]: Is this one used?

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on 6483d07737ac505259cb4b3383375b6dadf34089<br /><br />Could the icon name be in the manifest xml file? Or in config.xml as a hidden field?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31392">issues.joomla.org/tracker/joomla-cms/31392</a>.</sub>

File: administrator/language/en-GB/com_cpanel.ini (hunk 1 above)
Comment (Bakual) [reply administrator/language/en-GB/com_cpanel.ini:12]: I haven't found any use of that, that's why I deleted it (mentioned in the PR description).

**Comment by Bakual** [comment]
> Could the icon name be in the manifest xml file? Or in config.xml as a hidden field?

Could be an idea, I'll have a look at it. Maybe even the title could be taken from there, would make things much more intuitive for sure.

**Comment by Bakual** [comment]
Did an alternativ PR which makes use of the manifest file for title and icon specifications: https://github.com/joomla/joomla-cms/pull/31397

**Comment by Bakual** [comment]
I'm closing this one, as the other basically became an extended version. 😄

**Pull request closed without merging**